scripts/dex-router/src/router/jupiter.ts

```diff
@@ -2,21 +2,36 @@ import { Router, RouterOutput } from "../types";
 import { PublicKey, TransactionInstruction } from "@solana/web3.js";
 import {
   createJupiterApiClient,
+  DefaultApi,
   Instruction as JupiterInstruction,
 } from "@jup-ag/api";
 
 const MAX_SLIPPAGE_BPS = 50;
+const BASE_PATH_QUICKNODE = "https://jupiter-swap-api.quiknode.pro";
 
 export class JupiterRouter implements Router {
   private chainId: string;
   private executor: PublicKey;
   private maxAccounts: number;
-  private jupiterClient = createJupiterApiClient();
+  private jupiterClient: DefaultApi;
 
-  constructor(chainId: string, executor: PublicKey, maxAccounts: number) {
+  constructor(
+    chainId: string,
+    executor: PublicKey,
+    maxAccounts: number,
+    apiKey?: string
+  ) {
     this.chainId = chainId;
     this.executor = executor;
     this.maxAccounts = maxAccounts;
+    if (apiKey) {
+      this.jupiterClient = createJupiterApiClient({
+        basePath: BASE_PATH_QUICKNODE,
```

Comment:
this is a bit inflexible in case we want to use another provider or quicknode changes the domain format. Maybe we can use `URL.parse` to separate the basePath and apiKey?